src/Security/Authentication/Core/src/IDataSerializer.cs

```diff
@@ -1,11 +1,15 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.Diagnostics.CodeAnalysis;
+
 namespace Microsoft.AspNetCore.Authentication
 {
     public interface IDataSerializer<TModel>
     {
         byte[] Serialize(TModel model);
+
+        [return: MaybeNull]
```

Comment:
FYI https://github.com/dotnet/roslyn/pull/45993

This is fine for now, but we could switch MaybeNull/AllowNull attributes to use ? before .NET 5

Reply:
The version of the compiler we have doesn't support this as yet.